UsbAccess/UsbAccess.cpp

```diff
@@ -508,6 +563,172 @@ namespace Plugin {
+        }   
+        
+        while (fgets(buffer, 32, fp) != NULL);
+
+        string File_system = buffer;
+
+        int ret = pclose(fp);
+        if (ret == -1) 
+        {
+            result = false;
+        }
+
+        if (!result)
+        {
+            ent["error"] = "u disk device not found";
+            LOGINFO("u disk device not found\n");
+        }
+        else 
+        {
+            blocksize = fs.f_frsize /1024;
+            ent["path"] = pathParam;
+            ent["FileSystem"] = File_system;
+            ent["size(MB)"] = blocksize;
+            ent["TotalBlocks"] = fs.f_blocks;
+            ent["AvailableBlocks"] =  fs.f_bavail;
+            ent["TotalSpace(MB)"] = (fs.f_blocks * blocksize) / 1024;
+            ent["UsedSpace(MB)"] =  ((fs.f_blocks - fs.f_bavail) * blocksize) / 1024;
+            ent["AvailableSpace(MB)"] = (fs.f_bavail * blocksize) / 1024;
+        }
+
+        arr.Add(ent);
+        response["contents"] = arr;
+        returnResponse(result);    
+
+    }
+
+    uint32_t UsbAccess::getMountedInfoWrapper(const JsonObject &parameters, JsonObject &response)
+    {
+        LOGINFOMETHOD();
+        bool result = true;
+        char buffer[32] = "";
+        std::list<string> paths;
+        result = getMounted(paths);
+        JsonArray arr;
+        JsonObject ent;
+        LOGINFO("paths count = %d\n", paths.size());
+
+
+        if(paths.size() != 0)
+        {
+            for_each(paths.begin(), paths.end(), [&](const string& it)
+            {
+                string name = "";
+                string command = "findmnt -n -o source ";
+                command = command + it;
+                FILE* fp = popen(command.c_str(), "r");
+                if (fp == NULL) 
+                {
+                    result = false;
+                }   
+                
+                while (fgets(buffer, 32, fp) != NULL);
+
+                name = buffer;
+
+                int ret = pclose(fp);
+                if (ret == -1) 
+                {
+                    result = false;
+                }
+                ent["name"] = name;
+                ent["path"] = it;
+                arr.Add(ent);
+            });
+        }
+        else
+        {
+            result = false;
+            ent["error"] = "u disk device not found";
+            LOGINFO("u disk device not found\n");
+            arr.Add(ent);
+        }
+        
+        response["contents"] = arr;
+
+        returnResponse(result);
+    }
+
+
+    uint32_t UsbAccess::getDeviceCountWrapper(const JsonObject& parameters, JsonObject& response)
+    {
+        LOGINFOMETHOD();
+        bool result = true;
+        char buffer[32] = "";
+        string command =  "lsblk -o NAME,TYPE | grep \"disk\" | grep \"sd[a-z]\" | wc -l";
+        FILE* fp = popen(command.c_str(), "r");
+        if (fp == NULL) 
+        {
+            result = false;
+        }   
+        
+        while (fgets(buffer, 32, fp) != NULL);
```

Comment:
Same as above comment. fp is used in logic event after check with NULL. This could cause crash.

Reply:
Done